presto-main/src/main/java/com/facebook/presto/execution/QueryQueueRuleFactory.java

```diff
@@ -87,56 +84,38 @@ public QueryQueueRuleFactory(QueryManagerConfig config, ObjectMapper mapper)
 
     private static void checkIsDAG(List<QueryQueueRule> rules)
     {
-        DirectedPseudograph<String, DefaultEdge> graph = new DirectedPseudograph<>(DefaultEdge.class);
+        SimpleGraph<String, DefaultEdge> graph = new SimpleGraph<>(DefaultEdge.class);
         for (QueryQueueRule rule : rules) {
             String lastQueueName = null;
             for (QueryQueueDefinition queue : rule.getQueues()) {
                 String currentQueueName = queue.getTemplate();
                 graph.addVertex(currentQueueName);
                 if (lastQueueName != null) {
-                    graph.addEdge(lastQueueName, currentQueueName);
+                    try {
+                        graph.addEdge(lastQueueName, currentQueueName);
+                    }
+                    catch (IllegalArgumentException e) {
+                        throw new IllegalArgumentException(format("Queues must form a tree. Rules contain a loop for %s", currentQueueName), e);
```

Comment:
Is "a loop" a cycle or that a node has more than one "parent"? I think the terminology may be wrong, here.

i.e.,  this has a loop/cycle:

```
a -> b 
b -> c 
c -> a
```

but this does not (it's a DAG but not a tree):

```
a -> b
b -> c
a -> c
```